dbos/_dbos_config.py

```diff
@@ -291,3 +443,73 @@ def set_env_vars(config: ConfigFile) -> None:
     for env, value in config.get("env", {}).items():
         if value is not None:
             os.environ[env] = str(value)
+
+
+def overwrite_config(provided_config: ConfigFile) -> ConfigFile:
+    # Load the DBOS configuration file and force the use of:
+    # 1. The database connection parameters (sub the file data to the provided config)
+    # 2. OTLP traces endpoints (add the config data to the provided config)
+    # 3. Custom setup steps (sub the file data to the provided config)
+    # ? Name: should we override it with the config file ?
+    # ? logs level: right now this code ignores log level from the config file.
```

Comment:
These should fork in a convo about "what's the new requirements from `dbos-config.yaml`". Given it should only be for cloud parameters (language, app name, start/setup commands), I think we should respect the log level the user set in their code. With respect to the name, I don't know how we'd use it inside the library today -- maybe project C? @kraftp. In which case we probably want it from the config file, overriding whatever is given in the code.